diet score

caitlink12 opened the pull request:
FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, FVCDVEG, FVCDJUI (diet_score dependent variables) updated to include master survey cycles from 2001-2018 
diet_score updated to include master survey cycles from 2001-2018 
diet_score_cat3 updated to include master survey cycles from 2001-2018
master survey cycles added to FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, FVCDVEG, FVCDJUI, diet_score and diet_score_cat3 in variables.csv

caitlink12 committed:
temp

caitlink12 committed:
diet score update

added in ICES surveys to diet score derived variable

caitlink12 committed:
update diet_score_cat3

updated function diet_score_cat3 to include master survey cycles

caitlink12 committed:
temp

caitlink12 committed:
diet score dependents

added in master survey cycles for required diet score variables to derive diet score

caitlink12 committed:
spacing

adjusted spacing between variable entries

caitlink12 committed:
diet score-dependent variables and diet score

included master survey cycles for diet score and diet-score dependent variables

yulric committed:
fix big diffs

yulric commented on inst/extdata/variable_details.csv:
```diff
@@ -1148,14 +1148,14 @@ FVC_6E,N/A,cont,"cchs2009_s, cchs2010_s, cchs2012_s",[FVC_6E],,cont,copy,N/A,Eat
+FVCDFRU,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs20013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADFRU, cchs2003_p::FVCCDFRU, cchs ...
+FVCDFRU,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs20013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADFRU, cchs2003_p::FVCCDFRU, cchs ...
+FVCDFRU,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs20013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADFRU, cchs2003_p::FVCCDFRU, cchs ...
+FVCDFRU,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs20013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADFRU, cchs2003_p::FVCCDFRU, cchs ...
```
Looks there's an extra 0 in the `2013_2014_i cycle`, its `20013_2014_i` and it should be `2013_2014_i`

caitlink12 replied:
@yulric fixed.

yulric commented on inst/extdata/variable_details.csv:
```diff
@@ -1164,20 +1164,20 @@ FVCDPOT,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_
+FVCDSAL,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs20013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADSAL, cchs2003_p::FVCCDSAL, cchs ...
+FVCDSAL,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs20013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADSAL, cchs2003_p::FVCCDSAL, cchs ...
+FVCDSAL,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs20013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADSAL, cchs2003_p::FVCCDSAL, cchs ...
+FVCDTOT,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_s, cchs2010_s, cchs2012_s, cchs2001_i, cchs2003_i, cchs2005_i, cchs2007_2008_i, cchs2009_2010_i, cchs2011_2012_i, cchs2013_2014_i, cchs2015_2016_i","cchs2001_p::FVCADTOT, cchs2003_p::FVCCDTOT, cchs2 ...
```
Looks there's an extra 0 in the 2013_2014_i cycle, its 20013_2014_i and it should be 2013_2014_i

caitlink12 replied:
@yulric fixed

yulric commented on inst/extdata/variables.csv:
```diff
@@ -155,13 +155,13 @@
-"FVCDCAR","Carrot consumption","Daily consumption - carrots (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m","cchs2001_p::FVCADCAR, cchs2003_p::FVCCDCAR, cchs2005_p::FVCEDCAR, cchs2015_2016_p::FVCDVORA, cchs2017_2018_p: ...
-"FVCDFRU","Fruit consumption","Daily consumption - fruit - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m","cchs2001_p::FVCADFRU, cchs2003_p::FVCCDFRU, cchs2005_p::FVCEDFRU, cchs2015_2016_p::FVCDVFRU, cchs2017_2018_p:: ...
+"FVCDCAR","Carrot consumption","Daily consumption - carrots (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m, cchs20013_2014_m, cchs2 ...
+"FVCDFRU","Fruit consumption","Daily consumption - fruit - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m, cchs20013_2014_m, cchs20 ...
```
Looks there's an extra zero in the `cchs2013_2014_m` cycle.

caitlink12 replied:
@yulric fixed

yulric replied:
@caitlink12 I'm still seeing the extra zero in these lines

caitlink12 replied:
@yulric should be fixed now

yulric commented on inst/extdata/variables.csv:
```diff
@@ -155,13 +155,13 @@
 "FVCDJUI","Juice consumption","Daily consumption - fruit juice (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m, cchs20013_2014_m, cc ...
 "FVCDPOT","Potato consumption","Daily consumption - potatoes (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m, cchs20013_2014_m, cchs ...
-"FVCDSAL","Salad consumption","Daily consumption - green salad - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m","cchs2001_p::FVCADSAL, cchs2003_p::FVCCDSAL, cchs2005_p::FVCEDSAL, cchs2015_2016_p::FVCDVGRN, cchs2017_20 ...
+"FVCDSAL","Salad consumption","Daily consumption - green salad - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m, cchs20013_2014_m,  ...
```
Looks there's an extra zero in the `cchs2013_2014_m` cycle.

caitlink12 replied:
@yulric fixed

yulric replied:
@caitlink12 I'm still seeing the extra zero in this line

caitlink12 replied:
@yulric should be fixed now

yulric commented on inst/extdata/variables.csv:
```diff
@@ -155,13 +155,13 @@
+"FVCDSAL","Salad consumption","Daily consumption - green salad - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m, cchs20013_2014_m,  ...
 "FVCDTOT","Total fruit/veg consumption","Daily consumptoin - total fruits and veg. - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m ...
-"FVCDVEG","Other veg consumption","Daily consumption other vegetables - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m","cchs2001_p::FVCADVEG, cchs2003_p::FVCCDVEG, cchs2005_p::FVCEDVEG, cchs2015_2016_p::FVCDVVEG, cchs ...
+"FVCDVEG","Other veg consumption","Daily consumption other vegetables - (D)","Continuous","cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p, cchs2009_m, cchs2010_m, cchs2012_m, cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2011_2012_m, cchs20013_2 ...
```
Looks there's an extra zero in the `cchs2013_2014_m` cycle.

caitlink12 replied:
@yulric fixed. Also all _i suffixes have been changed to _m

yulric replied:
@caitlink12 I'm still seeing the extra zero in this line

caitlink12 replied:
@yulric should be fixed now

yulric reviewed (changes requested):
The database names are not consistent between the variables and variable details sheet. The variable details sheet uses the `_i` suffix whereas the variables sheet uses the `_m` suffix. Can you fix that?

caitlink12 committed:
suffixes

all _i suffixes changed to _m
edited cchs20013_2014_m to cchs2013_2014_m

yulric reviewed (changes requested):
I'm still seeing the extra zeros in some lines. I've replied to the comments that have them.

caitlink12 committed:
temp

caitlink12 committed:
extra zero

removed extra zero in cchs2013_2014_m cycle name for FVCDCAR, FVCDFRU, FVCDJUI, FVCDPOT, FVCDSAL, FVCDTOT, FVCDVEG

rafdoodle committed:
Added cchs2017_2018_m and units to all FVC(D) and diet score variables; fixed all their databaseStart and variableStart info too

DougManuel commented:
Reviewed the FVC and diet_score variable changes (FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, FVCDVEG, FVCDJUI, diet_score, diet_score_cat3). No issues found. Checked for:

- **variableStart mapping correctness**: All pre-2007 cycle-letter mappings (FVCA/FVCC/FVCE prefixes), 2007-2014 defaults, and 2015+ renamed variables (FVCDVFRU, FVCDVGRN, FVCDVORA, FVCDVPOT, FVCDVVEG, FVCDVJUI) are correct for both PUMF and master cycles.
- **databaseStart consistency**: variables.csv and variable_details.csv databaseStart fields match for all 8 variables. All 9 master cycles (cchs2001_m through cchs2017_2018_m) present.
- **Dangerous [VAR] defaults**: No `[VAR]` defaults cross the 2015 naming boundary — 2015+ master cycles all have explicit mappings.
- **FVCDJUI/FVCDPOT _i to _m correction**: Correctly replaced deprecated `_i` suffix with `_m`. The `cchs20013_2014_i` typo was fixed.
- **units field**: "times/day" added consistently to all FVC variable rows in variable_details.csv.
- **Derived variables**: diet_score and diet_score_cat3 correctly reference harmonized variable names via `DerivedVar::`.

Note: This review covers only the FVC/diet_score variables. The variables.csv full-file rewrite (column reordering, new metadata columns) was not reviewed here.

DougManuel commented:
### Code review

Reviewed 9 derived FVC/diet variables and 30 raw FVC variables for PUMF and Master across 2001-2018 cycles.

#### L6 integration test: cross-cycle prevalence

Ran `rec_with_table()` against PUMF data for all 12 cycles:

No step changes at the 2014-2015 era boundary. The 2015+ variable renames (FVCDVFRU, FVCDVGRN, FVCDVORA, FVCDVPOT, FVCDVVEG, FVCDVJUI) are correctly mapped.

Master (`_m`) mappings validated by worksheet checks only -- no runtime data available for L6 testing.

#### Issues found and fixed

All issues below have been fixed in commits [`8939654`](https://github.com/Big-Life-Lab/cchsflow/commit/8939654) and [`d8f3890`](https://github.com/Big-Life-Lab/cchsflow/commit/d8f3890).

**P0 -- data bug:**
1. **FVC_6D swapped recEnd values**: valid range `[1,120]` was mapped to `NA::a` and not-applicable code `996` was mapped to `copy` -- reversed at runtime, producing wrong data

**P1 -- naming/mapping errors:**
2. **Database name typo in all 30 FVC_\* raw variables**: `chs2011_2012_m` and `chs2013_2014_m` should be `cchs2011_2012_m` and `cchs2013_2014_m` (missing leading `c`). Affected 174 rows across both worksheets.
3. **Dummy variable suffix-recEnd misalignment**: FVC_1A and FVC_5A had suffixes that didn't match their recEnd values (copy-paste error)
4. **Dummy variable naming**: renamed `_NA::a`/`_NA::b` to `_NAa`/`_NAb` for FVC_1A-6A and diet_score_cat3; set Func row dummyVariable to `N/A`

**P2 -- metadata quality:**
5. Fixed double spaces in FVC_4E, FVC_5E labels and catLabels
6. Fixed FVCDTOT "consumptoin" typo and trailing dash in labelLong
7. Fixed inconsistent labelLong formatting for FVCDFRU, FVCDSAL, FVCDVEG
8. Added descriptions for 7 FVCD\* derived frequency variables
9. Improved diet_score_cat3 labelLong and description
10. Added reviewNotes for 30 FVC_\* variables

**Schema:**
- Updated `metadata_registry.yaml` and `variable_details.yaml` dummyVariable regex to match actual convention (see #172)

Checked: era boundary defaults, databaseStart consistency (variables.csv vs variable_details.csv), PUMF/Master naming conventions, pre-2007 cycle letters, DV function specifications (diet_score_fun, diet_score_fun_cat), dummyVariable naming, swapped recEnd values, label/metadata consistency, known error patterns, and PUMF integration across all cycles.

**Note**: Post-approval commit 16a8f3ab was pushed on 2026-02-10 (after yulric's approval on 2025-12-04). Review fixes applied in commits 8939654 and d8f3890.

CEP: `ceps/cep-007-diet/`

DougManuel committed:
Fix worksheet errors found during PR #148 review

variables.csv:
- Fix chs→cchs typo in databaseStart for 30 FVC_* variables (2011/2013 master)
- Fix double spaces in FVC_4E, FVC_5E labels
- Fix FVCDTOT "consumptoin" typo and trailing dash in labelLong
- Fix inconsistent labelLong formatting for FVCDFRU, FVCDSAL, FVCDVEG
- Add descriptions for 7 FVCD* derived frequency variables
- Improve diet_score_cat3 labelLong and description
- Add reviewNotes for 30 FVC_* variables

variable_details.csv:
- Fix chs→cchs typo in databaseStart/variableStart for 144 FVC_* rows
- Fix FVC_6D swapped recEnd values (P0 data bug: valid range mapped to NA)
- Fix FVC_1A and FVC_5A dummy variable suffix-recEnd misalignment
- Rename _NA::a/_NA::b to _NAa/_NAb for FVC_1A-6A and diet_score_cat3
- Set diet_score_cat3 Func row dummyVariable to N/A
- Fix double spaces in FVC_4E, FVC_5E catLabel and related fields

DougManuel committed:
Update dummyVariable naming convention in schemas

- Fix metadata_registry.yaml: replace _NA::[a-z] with _NA[a-z] in regex,
  remove _cont{N} pattern (continuous variables use N/A), add missing
  category examples (_NAa, _NAb)
- Fix variable_details.yaml: update dummyVariable notes to match actual
  convention with regex pattern and correct examples

See issue #172 for propagation to other branches.

DougManuel reviewed (approved):
Approved. All issues from the review have been fixed in commits 8939654 and d8f3890 (see updated review comment above).

DougManuel committed:
Add CEP-007 diet review artifacts for PR #148

Review summary, L6 integration test script, PUMF test results,
and cross-cycle prevalence QMD from the diet-score PR review.
